src/Microsoft.JSInterop/IJSRuntime.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Represents an instance of a JavaScript runtime to which calls may be dispatched.
+    /// </summary>
+    public interface IJSRuntime
```

Comment:
Should we expose any sort of features/capabilities here? Did you plan to add that through more interfaces?

Reply:
Yes, the interfaces mechanism was working fine to distinguish cases where sync or unmarshalled interop is supported. Maybe that will be enough. If not we can figure out what extra capability advertisement would be appropriate when we have scenarios for it.